### PR TITLE
Cleaned up profile page fragment, added testing, linked friends list with database properly

### PR DESCRIPTION
**Pull request by Netzu00**

### CODE REVIEW

File: app/src/androidTest/java/com/github/scribeWizTeam/scribewiz/ProfilePageFragmentTest.kt [hunk 1]
```diff
@@ -22,6 +21,20 @@ class ProfilePageFragmentTest {
 
     @Before
```
Comment (christopherWilliams98) [inline app/src/androidTest/java/com/github/scribeWizTeam/scribewiz/ProfilePageFragmentTest.kt:22]: Nicely written tests. If you have time to add additional tests, it would perhaps be useful to test more in depth the behaviour of the app when a user adds a friend that does not exist in the database, or adds themself, or the same friend twice. otherwise looks good!

File: app/src/main/java/com/github/scribeWizTeam/scribewiz/fragments/ProfilePageFragment.kt
Comment (christopherWilliams98) [inline app/src/main/java/com/github/scribeWizTeam/scribewiz/fragments/ProfilePageFragment.kt:1]: Really nicely structured code, it was very easy to read and follow the code in the profile composable fragment and discern what it does.

Comment (christopherWilliams98) [inline app/src/main/java/com/github/scribeWizTeam/scribewiz/fragments/ProfilePageFragment.kt:1]: Nicely documented functions, despite them being private!

**Review by christopherWilliams98 — APPROVED** [review]
The looks well written and structed, and is now tested. I can reccomend more tests around corner cases when it comes to users adding friend ids, otherwise, the code runs fine and seems to not break anything else. Looks fine to merge?
